From f99dffb1ef7c8de67325e995b2ec9eb2e47ef753 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Fri, 5 Sep 2008 16:35:20 +0800
Subject: [PATCH 019/280] ENGR00090493 Fix MX37 VPU cause systeam crash in x16
 bit

Fix MX37 VPU cause systeam crash in x16 bit.
Add check whether it is MX51 case when write reg BIT_RUN_AUX_STD.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 11c48a2..ba25914 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -2219,10 +2219,12 @@ RetCode vpu_DecStartOneFrame(DecHandle handle, DecParam * param)
 		VpuWriteReg(CMD_DEC_PIC_START_BYTE, param->picStartByteOffset);
 	}
 
-	if (pDecInfo->openParam.bitstreamFormat == STD_DIV3)
-		VpuWriteReg(BIT_RUN_AUX_STD, 1);
-	else
-		VpuWriteReg(BIT_RUN_AUX_STD, 0);
+	if (cpu_is_mx51()) {
+		if (pDecInfo->openParam.bitstreamFormat == STD_DIV3)
+			VpuWriteReg(BIT_RUN_AUX_STD, 1);
+		else
+			VpuWriteReg(BIT_RUN_AUX_STD, 0);
+	}
 
 	BitIssueCommand(pCodecInst->instIndex, pCodecInst->codecMode, PIC_RUN);
 
-- 
1.8.0

